atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -392,8 +394,14 @@ private enum State {
         this.transactionOutcomeMetrics =
                 TransactionOutcomeMetrics.create(transactionMetrics, metricsManager.getTaggedRegistry());
         this.expectationsDataCollectionMetrics = ExpectationsMetrics.of(metricsManager.getTaggedRegistry());
+        this.locksManager = new TransactionLocksManager(timelockService);
+        closer.register(() -> {
+            // Not timed because tryUnlock() is an asynchronous operation.
+            traced("postCommitUnlock", locksManager::close);
+        });
         this.immutableTimestampLockManager = new ImmutableTimestampLockManager(
                 immutableTimestampLock, new DefaultLockValidityChecker(timelockService));
+        immutableTimestampLock.ifPresent(locksManager::registerLock);
```

Comment:
@fsamuel-bs I think this is what would allow us to remove this unlock from further upstream?

Reply:
Yup, I can put a PR on top of this one to clean-up the happy (https://github.com/palantir/atlasdb/blob/develop/atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java#L284).